### PR TITLE
Add summarization metrics [Part I of BA-4781]

### PR DESCRIPTION
**Pull request by cjllanwarne**
~~Early feedback requested on:~~

* ~~Would these metrics be useful?~~
* ~~Is there overhead of getting the metadata count?~~
  * ~~... and if so, is it worth it?~~
  * ~~... and if not, are the other metrics still worth capturing?~~

Ready for review!

### CODE REVIEW

File: services/src/test/scala/cromwell/services/metadata/impl/MetadataDatabaseAccessSpec.scala [hunk 1]
```diff
@@ -176,12 +187,7 @@ class MetadataDatabaseAccessSpec extends FlatSpec with Matchers with ScalaFuture
         workflow2Id <- baseWorkflowMetadata(Workflow2Name, Set(testLabel2, testLabel3))
 
         // refresh the metadata
-        _ <- dataAccess.refreshWorkflowMetadataSummaries(1000) map { summaryResult =>
-          withClue("summaryResult") {
-            summaryResult.increasingResult should be > 0L
-            summaryResult.decreasingResult should be >= 0L
-          }
-        }
+        _ <- dataAccess.refreshWorkflowMetadataSummaries(1000) map { assertRowsProcessedAndSummarizationComplete }
```
Comment (mcovarr) [inline services/src/test/scala/cromwell/services/metadata/impl/MetadataDatabaseAccessSpec.scala]: curlies can go, here and below

File: database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala [hunk 2]
```diff
@@ -208,7 +209,7 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
           },
         summaryName = summarizeNameIncreasing
       )
-    } yield maximumMetadataEntryId
+    } yield (maximumMetadataEntryId - previousMaxMetadataEntryId, totalTableSize.toLong - maximumMetadataEntryId)
```
Comment (kshakir) [inline database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala]: The second calculation is comparing apples and oranges.

- The `totalTableSize` is the number of rows in the table.
- The `maximumMetadataEntryId` is the max primary key id.

Check out these values in alpha.

```shell
$ <toolbox-connect> -e alpha "
SELECT TABLE_ROWS FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = 'METADATA_ENTRY';
SELECT SUMMARY_POSITION FROM SUMMARY_STATUS_ENTRY WHERE SUMMARY_NAME = 'WORKFLOW_METADATA_SUMMARY_ENTRY_INCREASING';
SELECT MAX(METADATA_JOURNAL_ID) FROM METADATA_ENTRY;
"
TABLE_ROWS
2818962057
SUMMARY_POSITION
3543925173
MAX(METADATA_JOURNAL_ID)
3543925179
```

Btw, the `select count(1)` generated by slick is going to be _slow_, probably timing out.
- https://dba.stackexchange.com/questions/130780/mysql-count-performance/130851
- http://slick.lightbend.com/doc/3.2.3/queries.html#aggregation
- https://stackoverflow.com/questions/5179969/what-is-better-in-mysql-count-or-count1#answer-5180023

TL;DR: You probably want to replace checking the table size with a query to get the max id in `METADATA_ENTRY`.

**Comment by cjllanwarne** [comment]
Going to refactor this a little
